tools/topology/topology2/get_abi.sh

```diff
@@ -11,6 +11,24 @@ cat <<EOF_HEADER
 Object.Base.manifest."sof_manifest" {
 	Object.Base.data."SOF ABI" {
 EOF_HEADER
-printf '\t\tbytes\t"0x%02x,' "$ABI_MAJOR"
-printf "0x%02x," "$ABI_MINOR"
-printf '0x%02x\"\n\t}\n}' "$ABI_PATCH"
+
+
+print_1byte_abi()
+{
+	printf '\t\tbytes\t"0x%02x,' "$1"
+	printf "0x%02x," "$2"
+	printf '0x%02x\"\n\t}\n}' "$3"
+}
+
+print_2bytes_abi()
+{
+	printf '\t\tbytes\t"0x%02x,0x%02x,' "$(("$1" & 0xff))" "$(("$2" & 0xff00))"
```

Comment:
$2 looks like a typo here.

You could avoid a fair amount of repetition and typos like these with my `print_two_bytes()` suggestion, not sure why you don't like it?

Alternatively:

```suggestion
    printf '\t\tbytes\t"0x%02x,0x%02x,' "$((ABI_MAJOR & 0xff))" "$((ABI_MAJOR & 0xff00))"
```
etc.

Reply:
Wait, this is also missing the `>> 8` shift, isn't it?

Are you testing this or just floating ideas? Either is fine but just let me know.

Reply:
Ahh yes, testing with the current abi isnt helping. I moddifed the ABI to 3:21:256 and it works fine now. Thanks!